test/full_node.js

```diff
@@ -0,0 +1,50 @@
+var Trust = require('../');
+var bcoin = require('bcoin');
+var testHelpers = require('./helpers');
+var consensus = require('bcoin/lib/protocol/consensus');
+var sinon = require('sinon');
+var should = require('should');
+require('should-sinon');
+
+describe('FullNode', () => {
+  var node = null;
+  var walletDB = null;
+  sinon.spy(Trust.TrustIsRisk.prototype, 'addTX');
+
+  beforeEach(() => testHelpers.getNode().then((n) => {
+    node = n;
+  }));
+
+  beforeEach(() => testHelpers.getWalletDB(node).then((w) => {
+    walletDB = w;
+  }));
+
+  afterEach(() => walletDB.close());
+  afterEach(() => node.close());
+
+  it('should be a bcoin instance', () => {
+    should(node).be.an.instanceof(bcoin.fullnode);
+  });
+
+  it('should call trust.addTX() on every transaction', async function() {
+    var sender = await testHelpers.getWallet(walletDB, 'sender');
+    var receiver = await testHelpers.getWallet(walletDB, 'receiver');
+
+    // Produce a block and reward the sender, so that we have a coin to spend.
+    await testHelpers.mineBlock(node, sender.getAddress('base58'));
+
+    // Make the coin spendable.
+    consensus.COINBASE_MATURITY = 0;
+
+    await testHelpers.time(100);
```

Comment:
We need this timeout because even though the node has been mined at this point, the next call will fail with "not enough balance" if we don't wait long enough.

We need the next timeout because the 'tx' event doesn't fire until after a few milliseconds on the `bcoin.fullnode` object.

Sleeping for 0 milliseconds would run all the events that are currently in the event loop and return to the test right after. This is not sufficient however, as the event handlers for those events may emit more events that will be added to the end of the event queue, and therefore processed after the test finishes.